jetty-websocket/websocket-javax-client/src/main/java/org/eclipse/jetty/websocket/javax/client/internal/JavaxWebSocketClientContainer.java

```diff
@@ -271,4 +274,55 @@ private ClientEndpointConfig getAnnotatedConfig(Object endpoint) throws Deployme
 
         return new AnnotatedClientEndpointConfig(anno);
     }
+
+    @Override
+    protected void doStart() throws Exception
+    {
+        doClientStart();
+        super.doStart();
+    }
+
+    protected void doClientStart()
+    {
+        try
+        {
+            Object context = getClass().getClassLoader()
+                .loadClass("org.eclipse.jetty.server.handler.ContextHandler")
+                .getMethod("getCurrentContext")
+                .invoke(null);
+
+            Object contextHandler = context.getClass()
+                .getMethod("getContextHandler")
+                .invoke(context);
+
+            AbstractLifeCycleListener shutdownListener = new AbstractLifeCycleListener()
+            {
+                @Override
+                public void lifeCycleStopping(LifeCycle event)
+                {
+                    LifeCycle.stop(JavaxWebSocketClientContainer.this);
+                }
+            };
+
+            contextHandler.getClass()
+                .getMethod("addEventListener", EventListener.class)
+                .invoke(contextHandler, shutdownListener);
```

Comment:
This may work, but it's not optimal for `dump()` purposes.
I would rather call (via reflection) `contextHandler.addBean(this)`.

Reply:
I can do this but I think we will still need this listener to remove the bean when stopping. Otherwise restarting the webapp will accumulate these client container beans on the `ContextHandler`.

Reply:
@lachlan-roberts fair enough with regards to removing the bean, but then I would prefer to have it symmetrical in `doClientStop()` -- i.e. call (via reflection) `contextHandler.removeBean(this)`.

Reply:
@lachlan-roberts fair enough with regards to removing the bean, but then I would prefer to have it symmetrical in `doClientStop()` -- i.e. call (via reflection) `contextHandler.removeBean(this)`.